### PR TITLE
Browser-like back and forth buttons

### LINKED ISSUE #50: Browser-like back and forth buttons
Problem: going back and forth between notes with back and forwards buttons like on a web browser would be super useful, particularly when traversing on a set of semantic links from the sidebar.

Proposed implementation:
- in FileEditorContainer.tsx, selectedFilePath should be watched and any useEffect change should update the history object also a state variable in FileEditorContainer.tsx
- Back and forward buttons in the title bar 
- Click and hold on the back/forward button to reveal dropdown list of past files navigated to and from

Idea proposed here: #32

### PR DESCRIPTION
**Pull request by uuur86**
Fixes #50

Currently, it seems to be working smoothly. We can test it and then roll it out. The only issue at the moment is that the history gets cleared when it's rendered by an upper layer. This will be addressed later by adding the persistence storage feature.

### CODE REVIEW

**Comment by samlhuillier** [comment]
This is really great. One suggestion I have is to move this component to the titlebar. This will make the design cleaner and also give us the benefit of being able to go back and forth when navigating from search results.

**Comment by uuur86** [comment]
Great idea! I'll move the component to the titlebar for a cleaner look and easier navigation.

**Comment by uuur86** [comment]
Sorry, I forgot to inform you. Currently, it seems to be working smoothly. It's ready to test.

**Comment by samlhuillier** [comment]
Screenshot 2024-03-29 at 21 04 10
Thank you for moving it up to the title bar! Small problem is on mac: it's overlapping with the system control buttons - I suggest you do something like if process.platform === "darwin" add a margin like 30px to cover it. I can adjust the margin amount properly afterwards on my mac if you don't have a mac to test this on

**Comment by samlhuillier** [comment]
Screenshot 2024-03-30 at 13 56 09
Looks good! Small nitpick is to make the background colour of the two button componentes be the same as the title bar - I think that could make it look slightly cleaner.

File: src/components/File/FileSideBar/FileHistoryBar.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        >
+          <ul style={{ listStyleType: "none", padding: "0", margin: "0" }}>
+            {menuChild.map((path, ind) => (
+              <li
+                key={ind + startInd}
+                style={{ padding: "4px 2px", cursor: "pointer" }}
+              >
+                <div
+                  key={ind + startInd}
+                  onClick={() => goIndex(ind + startInd)}
```
[the remark this answers is not in the document]
Comment (uuur86) [reply src/components/File/FileSideBar/FileHistoryBar.tsx:132]: of course.. you right.

**Comment by uuur86** [comment]
image
image
image
image
Does it look good?

File: src/components/TitleBar.tsx [hunk 2]
```diff
@@ -40,14 +43,22 @@ const TitleBar: React.FC<TitleBarProps> = ({
...
       style={{ backgroundColor: "#303030" }}
     >
+
       <div
-        className=" flex"
+        className="flex"
         style={
           platform === "darwin"
             ? { marginLeft: "70px" }
             : { marginLeft: "10px" }
```
Comment (samlhuillier) [inline src/components/TitleBar.tsx:52]:
```suggestion
            : { marginLeft: "2px" }
```

Comment (samlhuillier) [reply src/components/TitleBar.tsx:52]: Screenshot 2024-03-30 at 21 42 20
Small suggestion to make the left button be centered with the sidebar :)

File: src/components/TitleBar.tsx [hunk 3]
```diff
@@ -40,14 +43,22 @@ const TitleBar: React.FC<TitleBarProps> = ({
       className={`h-titlebar  flex justify-between`}
       style={{ backgroundColor: "#303030" }}
     >
+
       <div
-        className=" flex"
+        className="flex"
         style={
           platform === "darwin"
             ? { marginLeft: "70px" }
```
Comment (samlhuillier) [inline src/components/TitleBar.tsx:51]:
```suggestion
            ? { marginLeft: "60px" }
```

Comment (samlhuillier) [reply src/components/TitleBar.tsx:51]: Small suggestion to move the nav buttons closer to the system control buttons :)

**Comment by samlhuillier** [comment]
This is looking great! Added two small suggestions for the margins - but beyond that, happy to merge this in.

**Review by samlhuillier — APPROVED** [review]
Looks good! Thank you for working on this.

**Pull request merged**